src/Validator.js

```diff
@@ -0,0 +1,62 @@
+import { LOTTO } from './constant/index.js';
+import { ERROR_MESSAGE } from './constant/error.js';
+
+class Validator {
+  purchaseAmount(amount) {
+    this.#validateNaturalNumber(amount);
+    this.#validateThousandUnit(amount);
+  }
+
+  winningNumbers(numbers) {
+    numbers.forEach((num) => {
+      this.#validateNaturalNumber(num);
+      this.#validateRange(num, LOTTO.START_NUMBER, LOTTO.END_NUMBER);
+    });
+    this.#validateExactCount(numbers, LOTTO.COUNT);
+    this.#validateNoDuplicates(numbers);
+  }
+
+  bonusNumber(bonusNumber, winningNumbers) {
+    this.#validateNaturalNumber(bonusNumber);
+    this.#validateRange(bonusNumber, LOTTO.START_NUMBER, LOTTO.END_NUMBER);
+    this.#validateNoItemContained(bonusNumber, winningNumbers, '당첨 번호');
+  }
+
+  #validateNaturalNumber(value) {
+    if (!Number.isInteger(value) || value <= 0) {
+      throw new Error(ERROR_MESSAGE.NOT_NATURAL_NUMBER);
+    }
+  }
+
+  #validateThousandUnit(value) {
+    if (value % 1000 !== 0) {
+      throw new Error(ERROR_MESSAGE.NOT_THOUSAND_UNIT);
+    }
+  }
+
+  #validateRange(value, min, max) {
+    if (value < min || value > max) {
+      throw new Error(`${ERROR_MESSAGE.NOT_VALID_RANGE} (${min}~${max})`);
+    }
+  }
+
+  #validateNoDuplicates(arr) {
+    if (new Set(arr).size !== arr.length) {
+      throw new Error(ERROR_MESSAGE.DUPLICATED);
+    }
+  }
+
+  #validateExactCount(arr, count) {
+    if (arr.length !== count) {
+      throw new Error(ERROR_MESSAGE.NOT_EXACT_COUNT(count));
+    }
+  }
+
+  #validateNoItemContained(item, arr, arrName) {
+    if (arr.includes(item)) {
+      throw new Error(ERROR_MESSAGE.ITEM_CONTAINED(arrName));
+    }
+  }
```

Comment:
클래스 내의 메서드가 `this` 를 사용하지 않는다면 static으로 선언해야 한다는 airbnb 코드 컨벤션이 있어서 저는 유틸리티 함수로 validate 함수들을 뺐습니다! 

`this`를 사용하지 않는 private 메서드는 static  private 으로 선언하는건 어떨까요?

https://github.com/airbnb/javascript?tab=readme-ov-file#classes--methods-use-this

```suggestion
  static #validateNoItemContained(item, arr, arrName) {
    if (arr.includes(item)) {
      throw new Error(ERROR_MESSAGE.ITEM_CONTAINED(arrName));
    }
  }
```

Reply:
메서드에는 private static 을 사용할 수 없는 것으로 알고 있습니다!
`class-methods-use-this` 규칙에 따르면 모두 static method로 선언해야하는 걸 알지만,
저는 Validator의 클래스 특성을 더 고려했던 것 같습니다! 다른 클래스와 달리, static만 호출하는 클래스라는 걸 명시해주고 싶었습니다!
그런데, @DonggyunKim00 님 말씀을 듣고 곰곰히 생각해보니 이럴거면 class가 아니라 함수로 만들었어야됐다는 생각이 드네요!
좋은 피드백 감사합니다 :)